examples/graphbolt/disk_based_feature/node_classification.py

```diff
@@ -0,0 +1,518 @@
+    )
+    parser.add_argument("--early-stopping-patience", type=int, default=25)
+    parser.add_argument(
+        "--sample-mode",
+        default="sample_neighbor",
+        choices=["sample_neighbor", "sample_layer_neighbor"],
+        help="The sampling function when doing layerwise sampling.",
+    )
+    parser.add_argument("--precision", type=str, default="high")
+    parser.add_argument("--enable-inference", action="store_true")
+    return parser.parse_args()
+
+
+def main():
+    torch.set_float32_matmul_precision(args.precision)
+    if not torch.cuda.is_available():
+        args.mode = "cpu-cpu-cpu"
+    print(f"Training in {args.mode} mode.")
+    args.graph_device, args.feature_device, args.device = args.mode.split("-")
+    args.overlap_feature_fetch = args.feature_device == "pinned"
+    # For now, only sample_layer_neighbor is faster with this option
+    args.overlap_graph_fetch = (
+        args.sample_mode == "sample_layer_neighbor"
+        and args.graph_device == "pinned"
+    )
+
+    # Load and preprocess dataset.
+    print("Loading data...")
+    disk_based_feature_keys = None
+    if args.cpu_cache_size_in_gigabytes > 0:
+        disk_based_feature_keys = [("node", None, "feat")]
+
+    dataset = gb.BuiltinDataset(args.dataset, root=args.root)
+    if disk_based_feature_keys is None:
+        disk_based_feature_keys = set()
+    for feature in dataset.yaml_data["feature_data"]:
+        feature_key = (feature["domain"], feature["type"], feature["name"])
+        # Set the in_memory setting to False without modifying YAML file.
+        if feature_key in disk_based_feature_keys:
+            feature["in_memory"] = False
+    dataset = dataset.load()
+
+    # Move the dataset to the selected storage.
+    graph = (
+        dataset.graph.pin_memory_()
+        if args.graph_device == "pinned"
+        else dataset.graph.to(args.graph_device)
+    )
+    features = (
+        dataset.feature.pin_memory_()
+        if args.feature_device == "pinned"
+        else dataset.feature.to(args.feature_device)
+    )
+
+    train_set = dataset.tasks[0].train_set
+    valid_set = dataset.tasks[0].validation_set
+    test_set = dataset.tasks[0].test_set
+    all_nodes_set = dataset.all_nodes_set
+    args.fanout = list(map(int, args.fanout.split(",")))
+
+    num_classes = dataset.tasks[0].metadata["num_classes"]
+
+    if args.cpu_cache_size_in_gigabytes > 0 and isinstance(
+        features[("node", None, "feat")], gb.DiskBasedFeature
+    ):
+        features[("node", None, "feat")] = gb.CPUCachedFeature(
+            features[("node", None, "feat")],
+            int(args.cpu_cache_size_in_gigabytes * 1024 * 1024 * 1024),
+            args.cpu_feature_cache_policy,
+            args.feature_device == "pinned",
+        )
+        cpu_cached_feature = features[("node", None, "feat")]
+        cpu_cache_miss_rate_fn = lambda: cpu_cached_feature._feature.miss_rate
+    else:
+        cpu_cache_miss_rate_fn = lambda: 1
```

Comment:
Let's highlight these sections and tell the user what these are for and what are the pitfalls of using this feature. See other examples to see how to write Highlight comment blocks.

Reply:
Is this the highlight comment block? https://github.com/dmlc/dgl/blob/master/examples/graphbolt/node_classification.py#L55-L69

Reply:
Yes, basically we want to leave comments to explain or give information about these new features the user sees for the first time.